api/proto/metrics/metrics.proto

```diff
@@ -24,6 +24,42 @@ message memoryMetric {
   double availPercentage = 5;
 }
 
+// App Memory Metric
+message AppMemoryMetric {
+    // total_MB - Memory allocated to app instance by EVE. In Steady State,
+    //  this is expected to be same as in App instance config. Eve should
+    //  report what it actually allocated / reserved instead of what is configured.
+    uint32 total_MB = 1;
+
+    // used_MB - Memory Used by app from total_MB Memory
+    //  used_MB <= total_MB
+    uint32 used_MB = 2;
+
+    // avail_MB - Available memory remaining out of Total for the App.
+    //  total_MB >= used_MB + avail_MB
+    uint32 avail_MB = 3;
+}
+
+message DeviceMemoryMetric {
+  // total_mem_MB - Total Device memory as seen by EVE.
+  uint32 total_mem_MB = 1;
+
+  // allocated_mem_apps_MB - Total memory Allocated to AppInstances
+  uint32 allocated_mem_apps_MB = 2;
+
+  // allocated_eve_MB - Total Allocated Memory for Eve
+  // This is the Max available memory for EVE on the device.
+  uint32 allocated_eve_MB = 3;
+
+  // used_mem_eve_MB - Memory currently used by Eve.
+  //    used_mem_eve_MB <= used_mem_eve_MB
+  uint32 used_mem_eve_MB = 4;
+
+  // available_mem_new_apps_MB - Memory Available for New Apps.
+  //    available_mem_new_apps_MB = total_mem_MB - (allocated_mem_apps_MB + allocated_eve_MB)
+  uint32 available_mem_new_apps_MB = 5;
```

Comment:
The semantics of this will have to be explained. IOW, why isn't it redundant?

Reply:
It is redundant in most cases.. Wanted Eve to explicitly say how much memory is available for new Apps - instead of controller calculating it.

For example, in cases where Eve usage is > allocated to Eve, this number can be lower than the above calculation.

Changed it to:

// available_new_appsMB - Memory Available for New Apps.
  // Generally,
  //    available_new_appsMB = totalMB - (allocated_appsMB + allocated_eveMB)
  // But in cases where EVE used memory > Eve allocated memory or AppUsed > AppAllocated,
  // this can be calculated using the used values instead of allocated ones.

Reply:
After discussion with @nordmark , removed this field.